client/src/modules/TeamDistribution/components/TeamDistributionCard/Actions.tsx

```diff
@@ -0,0 +1,104 @@
+import { Button, Modal, Row, Space, Typography } from 'antd';
+import { DownOutlined } from '@ant-design/icons';
+import { TextProps } from 'antd/lib/typography/Text';
+import { TeamDistributionDto, TeamDistributionDtoRegistrationStatusEnum } from 'api';
+import { dateWithTimeZoneRenderer } from 'components/Table';
+import moment from 'moment';
+
+const { Text, Link } = Typography;
+
+type Props = {
+  distribution: TeamDistributionDto;
+  onRegister: (distributionId: number) => Promise<void>;
+  onDeleteRegister: (distributionId: number) => Promise<void>;
+};
+
+const getDateColor = (date: string): TextProps['type'] => {
+  const now = moment();
+  const currentDate = moment(date);
+
+  const isDeadlineSoon = now <= currentDate && currentDate.diff(now, 'hours') < 48;
+
+  if (isDeadlineSoon) return 'danger';
+};
+
+export function Actions({ distribution, onRegister, onDeleteRegister }: Props) {
+  const timezone = Intl.DateTimeFormat().resolvedOptions().timeZone;
+
+  const handleCancel = async () => {
+    Modal.confirm({
+      title: 'Cancel registration',
+      content: (
+        <>
+          Are you sure you want to cancel your group task registration? You will be able to register again until{' '}
+          {dateWithTimeZoneRenderer(timezone, 'YYYY-MM-DD HH:mm')(distribution.endDate)}.
```

Comment:
Seems like you can extract it to separate function or variable and reuse it below